Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -278,6 +278,50 @@ void ReflectometryReductionOne::init() {
                   Direction::Input);
 }
 
+//----------------------------------------------------------------------------------------------
+/** Validate the input properties.
+*/
+std::map<std::string, std::string> ReflectometryReductionOne::validateInputs() {
+
+  std::map<std::string, std::string> result;
+
+  MatrixWorkspace_sptr decWS = getProperty("DetectorEfficiencyCorrection");
+  MatrixWorkspace_sptr runWS = getProperty("InputWorkspace");
+
+  if (!decWS) {
+    return result;
+  }
+
+  if (decWS->getAxis(0)->unit()->unitID() != "Wavelength") {
+    result["DetectorEfficiencyCorrection"] = "Detector Efficiency Correction "
+      "workspace x-units must be in wavelength";
+  }
+  else if (decWS->getNumberHistograms() > 1 &&
+      decWS->getNumberHistograms() < runWS->getNumberHistograms()) {
+    result["DetectorEfficiencyCorrection"] = "The number of spectra in the "
+      "Detector Efficiency Correction workspace must either be 1 or equal to "
+      "the number of spectra in the input workspace";
+  } else {
+    double firstVal = decWS->y(0)[0];
+    if (firstVal == 0) {
+      result["DetectorEfficiencyCorrection"] = "Detector Efficiency Correction "
+        "workspace cannot contain any zero values.";
+    } else {
+      for (size_t i = 1; i < decWS->getNumberHistograms(); i++) {
+        if (decWS->y(i)[0] != firstVal) {
+          result["DetectorEfficiencyCorrection"] = "All values in the Detector "
+            "Efficiciency Correction Workspace must be equal";
+          break;
```

Comment:
After meeting with instrument scientists, we decided on the following:

- Flood workspace should be checked if it is in units of wavelength and if not, convert it to units of wavelength.
- ConvertUnits to be carried out on input workspace before splitting it into detectors and monitors.
- After both conversions, divide input workspace by flood workspace.